server/channels/store/sqlstore/outgoing_oauth_connection_store.go

```diff
@@ -82,6 +85,10 @@ func (s *SqlOutgoingOAuthConnectionStore) GetConnections(c request.CTX, filters
 		query = query.Where("Id > ?", filters.OffsetId)
 	}
 
+	if filters.Audience != "" {
+		query = query.Where(sq.Like{"Audiences": fmt.Sprint("%", filters.Audience, "%")})
+	}
```

Comment:
This is what I was talking about yesterday, though there's a different method to get the connection by audience in [this PR](https://github.com/mattermost/enterprise/pull/1612/files#diff-254661b96d59861381f7aa5f57f84a92bc3ff4a2e4e3dae11393181cc1b89950R172)

Reply:
I had missed the tag, but is the reason that we didn't address it that we don't expect it to be used directly but rather via the `GetConnectionForAudience` which also does `audienceMatchesURL` and mitigates the issue that @mickmister highlighed?

Reply:
Yeah, the store is permissive so we can search using the basic `WHERE LIKE` query by whatever we need, but the list endpoint uses the `GetConnectionForAudience` method when trying to filter by audience, so that filter can only be used internally at the moment.